panther_gazebo/src/led_strip.cpp

```diff
@@ -0,0 +1,298 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_gazebo/led_strip.hpp"
+
+#include <gz/msgs/color.pb.h>
+#include <gz/msgs/marker.pb.h>
+#include <gz/plugin/Register.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/components/LightCmd.hh>
+#include <gz/sim/components/Name.hh>
+
+namespace panther_gazebo
+{
+
+LEDStrip::LEDStrip() : gz::sim::System() {}
+
+LEDStrip::~LEDStrip() = default;
+
+void LEDStrip::Configure(
+  const gz::sim::Entity & entity, const std::shared_ptr<const sdf::Element> & sdf,
+  gz::sim::EntityComponentManager & ecm, gz::sim::EventManager &)
+{
+  const auto model = gz::gazebo::Model(entity);
+  if (!model.Valid(ecm)) {
+    throw std::runtime_error(
+      "Error: Failed to initialize because [" + model.Name(ecm) +
+      "] (Entity=" + std::to_string(entity) +
+      ") is not a model. Please make sure that LEDStrip is attached to a valid model.");
+    return;
+  }
+
+  ParseParameters(sdf);
+  ConfigureLightEntityProperty(ecm);
+
+  node.Subscribe(ns + "/" + image_topic, &LEDStrip::ImageCallback, this);
+}
+
+void LEDStrip::PreUpdate(const gz::sim::UpdateInfo & info, gz::sim::EntityComponentManager & ecm)
+{
+  auto current_time = info.simTime;
+
+  auto period = std::chrono::milliseconds(static_cast<int>(1000 / frequency));
+  if (new_image_available && current_time - last_update_time >= period) {
+    gz::msgs::Image image;
+    {
+      std::lock_guard<std::mutex> lock(image_mutex);
+      image = last_image;
+      new_image_available = false;
+    }
+
+    VisualizeLights(ecm, image);
+
+    auto light_pose = ecm.Component<gz::sim::components::Pose>(light_entity)->Data();
+    VisualizeMarkers(image, light_pose);
+
+    last_update_time = current_time;
+  }
+}
+
+void LEDStrip::ImageCallback(const gz::msgs::Image & msg)
+{
+  try {
+    MsgValidation(msg);
+    std::lock_guard<std::mutex> lock(image_mutex);
+    last_image = msg;
+    new_image_available = true;
+  } catch (const std::exception & e) {
+    ignerr << "Error: " << e.what() << std::endl;
+  }
+}
+
+void LEDStrip::ParseParameters(const std::shared_ptr<const sdf::Element> & sdf)
+{
+  if (sdf->HasElement("light_name")) {
+    light_name = sdf->Get<std::string>("light_name");
+  } else {
+    throw std::runtime_error("Error: The light_name parameter is missing.");
+  }
```

Comment:
Let's reduce brackets and elses. If it is possible to throw an error in one if statement let's do this.
```suggestion
  if (!sdf->HasElement("light_name")) {
    throw std::runtime_error("Error: The light_name parameter is missing.");
  } 
  
  light_name = sdf->Get<std::string>("light_name");
```

Reply:
ok